pyanaconda/modules/timezone/timezone.py

```diff
@@ -0,0 +1,140 @@
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY expressed or implied, including the implied warranties of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.  You should have received a copy of the
+# GNU General Public License along with this program; if not, write to the
+# Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+# source code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission of
+# Red Hat, Inc.
+#
+from pyanaconda.dbus import DBus
+from pyanaconda.dbus.constants import MODULE_TIMEZONE_NAME, MODULE_TIMEZONE_PATH
+from pyanaconda.isignal import Signaled, Signal
+from pyanaconda.modules.base import KickstartModule
+from pyanaconda.modules.timezone.timezone_interface import TimezoneInterface
+from pyanaconda.modules.timezone.timezone_kickstart import TimezoneKickstartSpecification
+
+from pyanaconda import anaconda_logging
+log = anaconda_logging.get_dbus_module_logger(__name__)
+
+
+class TimezoneModule(KickstartModule):
+    """The Timezone module."""
+
+    def __init__(self):
+        super().__init__()
+        self._module_changed = Signal()
+        self._timezone = Signaled("America/New_York")
+        self._is_utc = Signaled(False)
+        self._ntp_enabled = Signaled(True)
+        self._ntp_servers = Signaled([])
+
+    def publish(self):
+        """Publish the module."""
+        DBus.publish_object(TimezoneInterface(self), MODULE_TIMEZONE_PATH)
+        DBus.register_service(MODULE_TIMEZONE_NAME)
+
+    def ping(self, s):
+        """Ping the module."""
+        log.debug(s)
+        return "TimeModule says hi!"
+
+    @property
+    def kickstart_specification(self):
+        """Return the kickstart specification."""
+        return TimezoneKickstartSpecification
+
+    def process_kickstart(self, data):
+        """Process the kickstart data."""
+        self.set_timezone(data.timezone.timezone)
+        self.set_utc(data.timezone.isUtc)
+        self.set_ntp(not data.timezone.nontp)
+        self.set_ntp_servers(data.timezone.ntpservers)
+
+    def generate_kickstart(self):
+        """Return the kickstart string."""
+        data = self.get_kickstart_data()
+        data.timezone.timezone = self.timezone
+        data.timezone.isUtc = self.is_utc
+        data.timezone.nontp = not self.ntp_enabled
+
+        if self.ntp_enabled:
+            data.timezone.ntpservers = list(self.ntp_servers)
+
+        return str(data)
+
+    @property
+    def module_changed(self):
+        """Signal that module might have changed.
+
+        If someone changes properties of this module
+        on the python level (and not from DBus), he
+        should emit this signal once he is done.
+        """
+        return self._module_changed
```

Comment:
I guess it could be part of the KickstartModule and KickstartModuleInterface.

About the name, I wanted to choose a different name, because there is ProperiesChanged from DBus, properties_changed from the cached observer and this signal and they all emit different values.

Reply:
Then go for `module_properties_changed`. I don't like `module_changed` because it is way too vague. If you start using it at first time you don't know what means that module changed.

Reply:
Fixed in a new commit.